lib/commons/standards/implicit-html-roles.js

```diff
@@ -58,7 +73,18 @@ const implicitHtmlRoles = {
     return vNode.hasAttr('href') ? 'link' : null;
   },
   article: 'article',
-  aside: 'complementary',
+  aside: vNode => {
+    if (
+      closest(vNode.parent, getSectioningContentSelector()) &&
+      // An aside within sectioning content can still be mapped to
+      // role=complementary if it has an accessible name
+      !hasAccessibleName(vNode)
```

Comment:
[Scott O'Hara's test page for aside mapping](https://scottaohara.github.io/testing/aside/) has an explicit test for the title attribute, so this code would fail his test page.

The motivating problem behind this pull request is the landmark-unique check. Ultimately we want good landmarks (distinct, not too many) for screen reader users. In that case, it would be better to flag nested asides with titles because in some browsers (Chrome) those will be given role=complementary, and therefore probably treated as landmarks by screen readers, potentially creating landmark ambiguity and noise.

For some other scenarios or checks, it might be more appropriate to only return the implicit role that the majority of browsers agree upon. 

This is why I did not modify the hasAccessibleName() function in my latest push to this pull request because I'm not sure it makes sense to do so for other accessibility checks. I only know it makes sense to do so for the landmark-unique check.

Should I open a PR that adds a check for title to the function? Then it could be discussed there.

Reply:
Thanks for looking into that. I tested out `title` attributes on `aside` and it looks like it works in all assistive technologies. So we'll probably want to allow `title` attribute to contribute to accessible name in these cases.

However I also tested out `section` with a `title` and VoiceOver on Safari still continues to not treat it as a region, so we'll not want to update the `hasAccessibleName` function but instead add another check to the if statement in the aside function to check if it has content in the `title` attribute (with an appropriate test)

Reply:
Thanks! That makes sense.

Just thinking out loud...would it make sense to add an options object to hasAccessibleName()? 

Then we could call `hasAccessibleName({ checkTitle: true })`...

Reply:
Sounds good to me